nomad/eval_broker.go

```diff
@@ -663,6 +699,12 @@ func (b *EvalBroker) Flush() {
 		wait.Stop()
 	}
 
+	// Cancel the delayed evaluations goroutine
+	b.delayedEvalCancelFunc()
```

Comment:
It is probably worth guarding this such that if the function is nil it doesn't panic:

```
if b.delayedEvalCancelFunc != nil {
```